### PR TITLE
Remove joblib and old-style samplers

### PR DESCRIPTION
**Pull request by krzentner**
Note that OffPolicyVectorizedSampler still exists, for now.

### CODE REVIEW

File: scripts/setup_linux.sh [hunk 1]
```diff
@@ -144,18 +144,18 @@ sudo apt install -y \
...
-git checkout 0be4f3f75aebd9d24583ee86590a38e741db0904
-mkdir glfw-build
-cd glfw-build
-sudo cmake -DBUILD_SHARED_LIBS=ON -DGLFW_BUILD_EXAMPLES=OFF \
-  -DGLFW_BUILD_TESTS=OFF -DGLFW_BUILD_DOCS=OFF ..
-sudo make -j"$(nproc)"
-sudo make install
-cd "${GARAGE_DIR}"
+# sudo apt purge -y libglfw*
+# GLFW_DIR="$(mktemp -d)/glfw"
```
Comment (ryanjulian) [inline scripts/setup_linux.sh:148]: this is needed on 18.04 and 16.04, but not 20.04 -- perhaps fork this into a new script?

Comment (krzentner) [reply scripts/setup_linux.sh:148]: Oh whoops. I didn't think I'd staged this file.

File: src/garage/experiment/experiment.py [hunk 2]
```diff
@@ -701,22 +701,28 @@ def make_launcher_archive(*, git_root_path, log_dir):
...
-    files_to_archive = subprocess.check_output(
+    git_files = subprocess.check_output(
         ('git', 'ls-files', '--others', '--exclude-standard', '--cached',
          '-z'),
         cwd=git_root_path).strip()
-    repo_size = sum([
-        os.stat(os.path.join(git_root_path, f)).st_size
-        for f in files_to_archive.split(b'\0')
-    ])
+    repo_size = 0
```
Comment (krzentner) [inline src/garage/experiment/experiment.py:708]: This stuff should actually be in a separate PR. It fixes `wrap_experiment` when there are deleted files in the current launch repo.

File: examples/np/cem_cartpole.py [hunk 3]
```diff
@@ -47,7 +47,7 @@ def cem_cartpole(ctxt=None, seed=1):
                    max_path_length=100,
                    n_samples=n_samples)
 
-        runner.setup(algo, env, sampler_cls=OnPolicyVectorizedSampler)
+        runner.setup(algo, env, sampler_cls=MultiprocessingSampler)
```
Comment (ryanjulian) [inline examples/np/cem_cartpole.py:50]: what would the default be if you removed this? is it not okay?

Comment (krzentner) [reply examples/np/cem_cartpole.py:50]: It is okay to remove this too. I think I'll replace it with `VecWorker` instead.

File: src/garage/experiment/experiment.py (hunk 2 above)
Comment (avnishn) [reply src/garage/experiment/experiment.py:708]: I have had this issue before. Very necessary

**Comment by ryanjulian** [comment]
@Mergifyio rebase

**Comment by ryanjulian** [comment]
This can now be merged

**Comment by ryanjulian** [comment]
I <3 this PR
